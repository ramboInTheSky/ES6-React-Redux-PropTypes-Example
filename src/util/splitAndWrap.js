import React from 'react';
import replace from 'ramda/src/replace';

// this function will accept an integer N, maximum characters
// and will shorten a file name by stripping out first N and last 3 characters
// i.e. shortenFileName(3)('some-big-title.jpg') => som...tle.jpg
export const shortenFileName = maxChars => {
  const regex = new RegExp(`(.{${maxChars}}).+(.{3})\\.`);
  return replace(regex, '$1...$2.');
};

const splitAndWrap = (stringToSplit, splitter, WrapperComponent) => {
  const a = stringToSplit.split(splitter);
  return a.map(item => <WrapperComponent key={item.toString()}>{item}</WrapperComponent>);
};

export default splitAndWrap;
