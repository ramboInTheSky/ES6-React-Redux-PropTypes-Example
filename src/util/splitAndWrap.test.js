import styled from 'styled-components';
import splitAndWrap, { shortenFileName } from './splitAndWrap';

describe('split a string ', () => {
  it('passing a space as splitter will wrap each element with a div', () => {
    const string = 'this is a string that include spaces';
    const Wrapper = styled.div({});
    expect(splitAndWrap(string, ' ', Wrapper)).toMatchSnapshot();
  });

  it('passing a null splitter will return the same string wrapped in the passed node', () => {
    const string = 'this is a string that include spaces';
    const Wrapper = styled.div({});
    expect(splitAndWrap(string, null, Wrapper)).toMatchSnapshot();
  });
});

describe('shortenFileName', () => {
  it('should shorten a file name that has length more than 15 characters', () => {
    const shorten = shortenFileName(7);
    expect(shorten('hello.jpg')).toEqual('hello.jpg');
    expect(shorten('hellowonderfulwonderfulworld.jpg')).toEqual('hellowo...rld.jpg');
  });
});
