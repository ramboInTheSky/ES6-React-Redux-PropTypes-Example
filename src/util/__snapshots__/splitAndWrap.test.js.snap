// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`split a string  passing a null splitter will return the same string wrapped in the passed node 1`] = `
Array [
  <styled.div>
    this is a string that include spaces
</styled.div>,
]
`;

exports[`split a string  passing a space as splitter will wrap each element with a div 1`] = `
Array [
  <styled.div>
    this
</styled.div>,
  <styled.div>
    is
</styled.div>,
  <styled.div>
    a
</styled.div>,
  <styled.div>
    string
</styled.div>,
  <styled.div>
    that
</styled.div>,
  <styled.div>
    include
</styled.div>,
  <styled.div>
    spaces
</styled.div>,
]
`;
